src/PrestaShopBundle/Command/ConfigCommand.php

```diff
@@ -0,0 +1,320 @@
+            throw new Exception($msg, self::STATUS_INVALID_ACTION);
+        }
+
+        $this->action = $action;
+
+        $this->initShopConstraint();
+
+        $this->initLanguage();
+    }
+
+    /**
+     * init possible shopconstraints
+     */
+    private function initShopConstraint(): void
+    {
+        if ($this->input->getOption('shopId') && $this->input->getOption('shopGroupId')) {
+            throw new Exception('Both shopId and shopGroupId cannot be defined', self::STATUS_INVALID_OPTIONS);
+        }
+        // init shopConstraint
+        try {
+            if ($this->input->getOption('shopGroupId')) {
+                $this->shopConstraint = ShopConstraint::shopGroup((int) $this->input->getOption('shopGroupId'));
+            } elseif ($this->input->getOption('shopId')) {
+                $this->shopConstraint = ShopConstraint::shop((int) $this->input->getOption('shopId'));
+            } else {
+                $this->shopConstraint = ShopConstraint::allShops();
+            }
+        } catch (Exception $e) {
+            $msg = sprintf('Failed initializing ShopConstraint: %s', $e->getMessage());
+            throw new Exception($msg, self::STATUS_FAILED_SHOPCONSTRAINT);
+        }
+    }
+
+    /**
+     * initialize language if the option was given
+     */
+    private function initLanguage(): void
+    {
+        $inputlang = $this->input->getOption('lang');
+        if (!$inputlang) {
+            return;
+        }
+
+        // all languages
+        $onlyActive = true;
+        $onlyShopId = null;
+        if (!is_null($this->shopConstraint->getShopId())) {
+            $onlyShopId = $this->shopConstraint->getShopId()->getValue();
+        }
+        $languages = $this->languageDataProvider->getLanguages($onlyActive, $onlyShopId);
+
+        $found = current(array_filter($languages, function (array $item) use ($inputlang) {
+            $key = is_numeric($inputlang) ? 'id_lang' : 'iso_code';
+
+            return isset($item[$key]) && $inputlang == $item[$key];
+        }));
+
+        if (!$found) {
+            throw new Exception('Invalid language', self::STATUS_INVALID_OPTIONS);
+        }
+
+        $this->idLang = (int) $found['id_lang'];
+    }
+
+    /**
+     * Get and print out one configuration value
+     */
+    private function get(): void
+    {
+        $key = $this->input->getArgument('key');
+
+        if (strpos($key, '*') !== false) {
+            if (!$this->idLang) {
+                throw new Exception('Using wildcards requires --lang option', self::STATUS_LANG_REQUIRED);
+            }
+            try {
+                $keys = array_filter($this->configuration->keys(), function (string $v) use ($key) {
+                    return fnmatch($key, $v);
+                });
+            } catch (NotImplementedException $e) {
+                throw new Exception('Not implemented yet', self::STATUS_NOT_IMPLEMENTED);
```

Comment:
```suggestion
```

Reply:
https://github.com/PrestaShop/PrestaShop/blob/901a073a6466642d20bc6440e83fc472bfa7ceb5/src/Adapter/Configuration.php#L63 does not give any message and the next catch wants one (and gives a specific return code also)

I think it's better to keep this. With the catch:
```
root@05ee984cbfb2:/var/www/html# bin/console prestashop:config get mymodule_* --lang fi

                                                                                                                        
 [ERROR] Not implemented yet                                                                                            
                                                                                                                        

root@05ee984cbfb2:/var/www/html# echo $?
8
```

without
```
root@05ee984cbfb2:/var/www/html# bin/console prestashop:config get mymodule_* --lang fi

                                                                                                                        
 [ERROR] Could not load all configuration keys:                                                                         
                                                                                                                        

root@05ee984cbfb2:/var/www/html# echo $?
9
```